### ISSUE TITLE
Request Scope Interceptor for GRPC not working

### ISSUE THREAD

**Issue by Reasno** [nestjs/nest#2489]
## Current behavior

Arguments for request scope Interceptor are not injected for GRPC endpoint. All arguments are undefined.

This only happens with the exact combination of Interceptor + GRPC + Request scope. 

## Input Code

Repo:
https://github.com/Reasno/nest-grpc-interceptor-request-scope-bug

This assertion won't pass:
https://github.com/Reasno/nest-grpc-interceptor-request-scope-bug/blob/d38846e2b07631454345cb99486114696ed42471/src/test.interceptor.ts#L13

## Expected behavior
Request scope interceptor works for GRPC endpoint.

## Environment

<pre><code>
Nest version: 6.3.1

 
For Tooling issues:
- Node version: 10
- Platform:  mac

Others:

</code></pre>

**Comment by Reasno** [comment]
Request scope interceptor for HTTP endpoint ✅
Default scope interceptor for GRPC endpint ✅
Request scope service injection for GRPC endpoint ✅
Request Scope Interceptor for GRPC endpoint ❌

**Comment by kamilmysliwiec** [comment]
Why `'TEST'` is a request-scoped provider? `scope: Scope.REQUEST` in combination with `useValue` doesn't make sense (value is const so it would be shared across requests anyway)

**Comment by Reasno** [comment]
The TEST provider was just a quick example. In real application, we were implementing request tracing. I thought this wouldn’t matter, but if it does, I can update repo to reproduce this bug with a real tracer.

**Comment by kamilmysliwiec** [comment]
Yes, please do, it would help a lot :)

**Comment by Reasno** [comment]
Repo updated! Hopefully this will do.

```ts
import * as uuid from 'uuid/v4';

@Module({
  controllers: [HeroController],
  providers: [{
      provide: 'REQUEST_ID',
      useFactory: uuid,
      scope: Scope.REQUEST
  }]
})
```

**Comment by kamilmysliwiec** [comment]
Thanks! It should be fixed in 6.4.1
